From 8824c431506067f7f12b1140994a0ffa97a97c7a Mon Sep 17 00:00:00 2001
From: Samuel Dionne-Riel <samuel@dionne-riel.com>
Date: Mon, 18 Jul 2022 01:32:11 -0400
Subject: [PATCH] mount: Don't overwrite devtmpfs

---
 etc/init.d/rcS | 1 -
 1 file changed, 1 deletion(-)

diff --git a/etc/init.d/rcS b/etc/init.d/rcS
index 9b3f318..0bdc285 100755
--- a/etc/init.d/rcS
+++ b/etc/init.d/rcS
@@ -13,7 +13,6 @@ echo /sbin/mdev > /proc/sys/kernel/hotplug
 # Previously in `/etc/profile`.
 
 mount -t sysfs none /sys
-mount -t tmpfs mdev /dev
 mkdir -p /dev/pts
 mount -t devpts devpts /dev/pts
 mount -t debugfs none /sys/kernel/debug/
-- 
2.35.1

